dynacred/src/group/groupContractCollection.ts

```diff
@@ -0,0 +1,341 @@
+                }
+            } else {
+                if (!groupContract.verify(undefined)) {
+                    throw new TypeError("The group contract verification failed.");
+                }
+            }
+
+            // check if the id is not already there
+            const existing: GroupContract[] = [];
+            this._collection.forEach((gd: GroupContract) => {
+                if (gd.id === groupContract.id) {
+                    existing.push(gd);
+                }
+            });
+
+            if (existing.length) {
+                groupContract.mergeSignoffs(existing[0]);
+                this._collection.set(groupContract.id, groupContract);
+            } else {
+                // there is a new proposed group contract; therefore, erase the current proposed group contract
+                this.removeProposedGroupContract();
+
+                this._collection.set(groupContract.id, groupContract);
+            }
+
+            // if groupContract is accepted; therefore, it becomes the current group contract
+            const numbPredecessor = groupContract.groupDefinition.predecessor.length;
+            if (!keepOnly && (numbPredecessor === 0 || (numbPredecessor > 0 && this.isAccepted(groupContract)))) {
+                this.currentGroupContract = groupContract;
+            }
+        } catch (e) {
+            throw e;
+        }
+    }
+
+    /**
+     *
+     * @param groupContract
+     * @returns {boolean} true if the collection has the group contract, otherwise false
+     */
+    has(groupContract: GroupContract): boolean {
+        return this._collection.has(groupContract.id);
+    }
+
+    /**
+     * Returns group contract by id
+     *
+     * @param id
+     * @returns {GroupContract}
+     */
+    get(id: string): GroupContract {
+        return this._collection.get(id);
+    }
+
+    getCurrentGroupContract(): GroupContract {
+        return this.currentGroupContract;
+    }
+
+    getProposedGroupContract(): GroupContract {
+        try {
+            for (const gc of Array.from(this._collection.values())) {
+                // avoid returning the genesis group contract
+                if (gc.groupDefinition.predecessor.length > 0 && !this.isAccepted(gc)) {
+                    return gc;
+                }
+            }
+
+            return undefined;
+        } catch (e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Get worldview from a specific group contract
+     *
+     * @param groupContract
+     * @returns an array of direct successor(s)
+     */
+    getWorldView(groupContract: GroupContract) {
+        return this.getChildren(groupContract);
+    }
+
+    /**
+     * Get worldview from a specific group contract
+     *
+     * @param groupContract
+     * @returns returns [gd] if there is no child to groupContract,
+     * returns [[gd,gd2], [gd,gd3]] if there is two children to groupContract
+     */
+    // getWorldView(groupContract: GroupContract) {
```

Comment:
remove unused code - except if it can be revived.

Reply:
I've let this version of the `getWorldView` method because it is the original one. Maybe it can be useful or should I erase it?